src/shims/unix/fs.rs

```diff
@@ -97,7 +111,9 @@ impl FileDescription for FileHandle {
                 .expect("failed to restore file position, this shouldn't be possible");
             res
         };
-        Ok(f())
+        let result = f();
+        ecx.write_byte_helper(result, dest)?;
+        Ok(())
```

Comment:
```suggestion
```

Just remove the `?;` from the previous line

Applies in lots of other places in this PR, too